src/Opaleye/Internal/HaskellDB/PrimQuery.hs

```diff
@@ -60,6 +60,7 @@ data BinOp      = (:==) | (:<) | (:<=) | (:>) | (:>=) | (:<>)
                 | (:->) | (:->>) | (:#>) | (:#>>)
                 | (:@>) | (:<@) | (:?) | (:?|) | (:?&)
                 | (:&&) | (:<<) | (:>>) | (:&<) | (:&>) | (:-|-)
+                | OpArrayIndex
```

Comment:
I really must protest once more, I hate to break the logical consistency that I outlined above without good reason. I would argue that it is neater to be consistent in that all unary operators, regardless of syntax, are found under `UnOp` and all binary operators, regardless of syntax, are found under `BinOp`.

If you feel strongly that this is not intuitive or is somehow less organized I will implement it the other way, but then I would suggest following renames:

`BinExpr` -> `InfixBinExpr`
`BinOp` -> `InfixBinOp`